multicluster/cmd/unlink.go

```diff
@@ -95,5 +96,9 @@ func newUnlinkCommand() *cobra.Command {
 	cmd.Flags().StringVar(&opts.namespace, "namespace", defaultMulticlusterNamespace, "The namespace for the service account")
 	cmd.Flags().StringVar(&opts.clusterName, "cluster-name", "", "Cluster name")
```

Comment:
With `unlink` we expect to pass a `clusterName` to know which link crd to delete. Would it be possible to get completions for it?

e.g I have 2 k3d clusters, `east` and `west`  and in `east` I have a link to the targeted west cluster. Here's how it looks:
```
$ kubectl --context=east get links --all-namespaces

NAMESPACE              NAME   AGE
linkerd-multicluster   west   14d
```

If I want to unlink the two when I'm in `east`, would it be possible to do `linkerd mc unlink --namespace <ns> --cluster-name [tab][tab]` and have `west` selected? Would be pretty sick but I'm not sure how the client-go discovery client works with crds.

Reply:
Hmm, yea it might be possible. It could be we could read the list of clusters that are present in the user's Kubeconfig and give out suggestions based on that. I can see if I can add that to this PR.

Reply:
Hmm, might not be a feasible thing to implement now. I thought we would have an easy way to cycle through the link resources themselves in a cluster (i.e by invoking the complete fn with the `link` resource, similar to what we have in `viz`). I think reading the kubeconfig could give us options that we might not want to consider (e.g I have 3 clusters but only 2 of them are linked). It's also possible the link doesn't have the same name as the cluster in the kubeconfig (k3d prefixes names with `k3d-` for example, so my link is `west` but the cluster's name is `k3d-west`).

It might be good to just keep the PR as is rather than overcomplicate it. My fingers won't fall off if I type out the `-cluster-name` flag values 😅

Reply:
Ahh that's a very good point. We could most certainly do autocompletion on `--cluster-name` if it is tied to the link resources in the cluster. I wasn't aware that this was the case. My impression was that the `--cluster-name` was the name of the cluster in the kubeconfig. Let me see if I can add that, and if it's complicated I'll add it to a follow up PR.